### PR TITLE
SGX: Clear additional flag on enclave entry

### PR DESCRIPTION
**Pull request by raoulstrackx**
An attacker could set both the AC flag in CR0 as in rflags. This causes the enclave to perform an AEX upon a misaligned memory access, and an attacker learns some information about the internal enclave state. 
The AC flag in rflags is copied from userspace upon an enclave entry. Upon AEX it is copied and later restored. This patch forces the rflag.AC bit to be reset right after an enter.

### CODE REVIEW

File: src/libstd/sys/sgx/abi/entry.S [hunk 1]
```diff
@@ -121,6 +121,15 @@ sgx_entry:
     fnstcw %gs:tcsls_user_fcw
 /*  reset user state */
     cld /* x86-64 ABI requires DF to be unset at function entry/exit */
+
+/*  making sure AC flag is not set in rflags */
+/*  avoid using the 'clac' instruction to be compatible with older compilers */
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S]: older *CPUs*

File: src/libstd/sys/sgx/abi/entry.S [hunk 2]
```diff
@@ -121,6 +121,15 @@ sgx_entry:
...
+    pushfq
+    popq %rcx
+    and $0xFFFFFFFFFFFBFFFF, %rcx
+    push %rcx
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S]: You can clear DF and AC in a single instruction. e.g. `andq $~0x40400, (%rsp)`

File: src/libstd/sys/sgx/abi/entry.S (hunk 1 above)
Comment (raoulstrackx) [reply src/libstd/sys/sgx/abi/entry.S]: clac was introduced in 5th generation Intel CPUs, SGX in 6th generation, so this isn't an issue

File: src/libstd/sys/sgx/abi/entry.S (hunk 2 above)
Comment (raoulstrackx) [reply src/libstd/sys/sgx/abi/entry.S]: agreed. Fixed :)

File: src/libstd/sys/sgx/abi/entry.S [hunk 3]
```diff
@@ -119,8 +119,12 @@ sgx_entry:
     mov %rbx,%gs:tcsls_tcs_addr
     stmxcsr %gs:tcsls_user_mxcsr
     fnstcw %gs:tcsls_user_fcw
+
 /*  reset user state */
-    cld /* x86-64 ABI requires DF to be unset at function entry/exit */
+/*    - DF flag: x86-64 ABI requires DF to be unset at function entry/exit */
+/*    - AC flag: AEX on misaligned memory accesses leaks side channel info */
+    andq $~0x40400, (%rsp)
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S]: Still need pushfq/popfq

Comment (raoulstrackx) [reply src/libstd/sys/sgx/abi/entry.S]: Of course!

**Review by jethrogb — APPROVED** [review]

**Comment by jethrogb** [comment]
cc @Goirad can you please update toolchain after this is merged and a new nightly is published?

**Pull request merged**